### ISSUE TITLE
Model Selection

### ISSUE THREAD

**Issue by cMancio00** [cMancio00/Super-Resolution#17]
We have a validation set that we should use for model selection.
To keep things simple we train the model just for few configuration of the model parameters, i.e.:

-  `num_channels = [16, 32, 64]`
- `num_res_block = [4, 8, 16]`

The models to be trained are 9. We can discard a model that does not decrement the loss wrt the previous model.

- [x]  After finding the best model, save a checkpoint.
- [x] Retrain the model (starting from the checkpoint) 
- [x] Save the training logs
